lib/system/iterators_1.nim

```diff
@@ -104,10 +104,12 @@ template dotdotImpl(t) {.dirty.} =
     ##
     ## See also:
     ## * [..<](#..<.i,T,T)
-    var res = a
-    while res <= b:
-      yield res
-      inc(res)
```

Comment:
There are people who overload ``inc`` and expect "dot dot" to work with their custom types.

Reply:
this PR won't affect them, that's a different proc. `dotdotImpl` is only instantiated for int64,int32,uint64,uint32 and `mixin inc` isn't there (and couldn't possibly be there, because `iterator `..`*(a, b: t): t {.inline.} =` is not generic).

```nim
when true:
  template inc*(a: var int32) =
    doAssert false # not called
    a = a + 1
  for i in int32(1)..int32(3):
    echo (i,)
```
=> custom `inc` not called